### ISSUE TITLE
Spell checker is active for partial words.

### ISSUE THREAD

**Issue by vithiri** [olivierkes/manuskript#166]
I am not certain if this is a feature or a bug, but the spell checker is active even when a word has not been completed.

If I type "paragraph", the spell checker will kick in and highlight the word at "parag" since it's not a proper word (yet).

Could a better behaviour be to only check words for spelling once a space or a special character has been added by the end of it?

**Comment by olivierkes** [comment]
Good point. I tried a few options, but wasn't able to find a satisfying one.

The relevant code is this:
https://github.com/olivierkes/manuskript/blob/3b1a3f7fdef2df8a8270703420dff884a0edd985/manuskript/ui/editors/basicHighlighter.py#L82-L87

`WORDS = '(?iu)([\w\']+)\W'` is the regex used to match words (`(?iu)` means: case insensitive, unicode). Now if I use `'(?iu)[\w\']+\W'`, so I add "followed by a non-word character" `\W` (code has to be slightly modified, see below), then it will work as desired when typing, but it won't spellcheck words at the end of line:

![image](https://user-images.githubusercontent.com/1192933/31768509-f29d555c-b4cf-11e7-904e-e6a0de22bf08.png)

(Code has to be modify like that:)

```python
WORDS = '(?iu)([\w\']+)\W'  # ← added a capturing group on the word
if hasattr(self.editor, "spellcheck") and self.editor.spellcheck:
    for word_object in re.finditer(WORDS, text):
        if self.editor._dict and not self.editor._dict.check(word_object.group(1)):  # ← spell check the group
```

Maybe somebody has a better idea?

**Comment by olivierkes** [comment]
> Maybe somebody has a better idea?

Well actually I do... I add a space at the end of every checked line, except if the text cursor is there:

https://github.com/olivierkes/manuskript/blob/ee058a309fbc6a8cf0b1fa0a10f605a528ecb1c9/manuskript/ui/editors/basicHighlighter.py#L84-L91

Please test and confirm whether that behavior is better or not :)

**Comment by vithiri** [comment]
The behavior is clearly much better now, but an observation would be that it extends the wavy line to include the word terminator as well, whether that's the tailing space or a non word character -- but seemingly only the first one.

![spellchecker](https://user-images.githubusercontent.com/3118182/31786171-dd09c59e-b507-11e7-88a0-2a89148bf411.png)

**Comment by olivierkes** [comment]
> it extends the wavy line to include the word terminator as well (...)

Yep, stupid mistake. Fixed :)

> Here is a word that is behaving strangely (...)

Previous commit should fix that too:

![test](https://user-images.githubusercontent.com/1192933/31793793-8b57e32a-b520-11e7-9bd1-0fa5cb1a7230.gif)

It showed me another issue: in some cases, right-click does not select the same word to suggest corrections:

![test](https://user-images.githubusercontent.com/1192933/31794089-67259370-b521-11e7-8d96-080ee138af23.gif)

Memo: I need to correct that here: `WordUnderCursor` does not allow for quotation mark `'` it seems: https://github.com/olivierkes/manuskript/blob/3b1a3f7fdef2df8a8270703420dff884a0edd985/manuskript/ui/views/textEditView.py#L407-L412

**Comment by vithiri** [comment]
I noticed that both FocusWriter and Gedit suffer from the same issue where the word will be underlined as misspelled if i.e. `shouldn'` is typed, so perhaps this behavior is simply inherent in the spell checking.

**Comment by olivierkes** [comment]
(That was closed in 0.5.0 but I forgot to close it)

**Comment by vithiri** [comment]
@gedakc @olivierkes Did this issue nestle its way back in again at some point?

**Comment by gedakc** [comment]
Good catch @vithiri.  I preformed a git bisect and it appears that the problem with the spell checker being active while typing a word came back in **0.7.0** with  commit 63b471e which fixed issue #283.

I will take a look at this.
